cedar-lean/Cedar/Validation/RequestEntityValidator.lean

```diff
@@ -0,0 +1,157 @@
+import Cedar.Validation.Validator
+import Cedar.Validation.Typechecker
+namespace Cedar.Validation
+
+open Cedar.Spec
+open Cedar.Data
+
+inductive RequestValidationError where
+| typeError (loc : String)
+
+abbrev RequestValidationResult := Except RequestValidationError Unit
+
+inductive EntityValidationError where
+| typeError (loc : String)
+
+abbrev EntityValidationResult := Except EntityValidationError Unit
+
+def instanceOfBoolType (b : Bool) (bty : BoolType) : Bool :=
+match b, bty with
+  | true, .tt => true
+  | false, .ff => true
+  | _, .anyBool => true
+  | _, _ => false
+
+def instanceOfEntityType (e : EntityUID) (ety : EntityType ) : Bool := ety == e.ty
+
+def instanceOfExtType (ext : Ext) (extty: ExtType) : Bool :=
+match ext, extty with
+  | .decimal _, .decimal => true
+  | .ipaddr _, .ipAddr => true
+  | _, _ => false
+
+def requiredAttributesPresent (r : Map Attr Value) (rty : Map Attr (Qualified CedarType)) (k : Attr) :=
+match rty.find? k with
+    | .some qty => if qty.isRequired then r.contains k else true
+    | _ => true
+
+def instanceOfType (v : Value) (ty : CedarType) : Bool := match v, ty with
+| .prim (.bool b), .bool bty => instanceOfBoolType b bty
+| .prim (.int _), .int => true
+| .prim (.string _), .string => true
+| .prim (.entityUID e), .entity ety => instanceOfEntityType e ety
+| .set s, .set ty => (s.elts.attach.map (λ ⟨v, _⟩ => instanceOfType v ty)).all id
```

Comment:
Why do a `map` and then `all id`?  I think the following simpler expressions should work:

```
s.elts.attach.all (λ ⟨v, _⟩ => instanceOfType v ty)
```